doc/port-add-del-dynamically/dynamic_port_add_del_hld.md

```diff
@@ -0,0 +1,305 @@
+# Scope                                                                                  
+This document describes the high level design of orchagent and the impact of creating/removing ports dynamically on other services. The design describes the current implementaion and suggestion to changes that needs to be implemented in order to fully support the dynamic create/remove of ports.
+
+## Relevant PRs
+[PR #7999 Allow cfggen to work on system without ports](https://github.com/Azure/sonic-buildimage/pull/7999)<br />
+[PR #1860 Remove buffer drop counter when port is deleted](https://github.com/Azure/sonic-swss/pull/1860)<br />
+[PR #1808 [swss]: Allow portsyncd to run on system without ports](https://github.com/Azure/sonic-swss/pull/1808)<br />
+[PR #2019 [orchagent] add & remove port counters dynamically each time port was added or removed](https://github.com/Azure/sonic-swss/pull/2019)<br />
+[PR #2022 Dynamic port configuration - add port buffer cfg to the port ref counter](https://github.com/Azure/sonic-swss/pull/2022)<br />
+
+## Design
+
+
+<a name="init-stage"></a>
+
+# Initialization stage 
+
+ 
+ ![Init stage](images/init_stage_diagram.png)
+ 
+- **Portsyncd** read port config db info and push it to App db and will set PortConfigDone on App db when finished.
+- **Portsorch** (orchagent) for every port added to the APP DB … will create port through SAI call and create also host interface for each time port is added to port APP table.
+- **Portsyncd** will receive netlink notification for each host interface that was created, and update an entry on state db
+- When all host interfaces are created **Portsyncd** is setting PortInitDone.
+
+
+### App DB flags:
+PortConfigDone – finished to configure ports on init
+PortInitDone – all host interfaces were created 
+
+Some services are waiting for these flags before they continue to run:
+Orchagent is waiting for PortConfigDone before continuing to create the ports on SAI.
+Xcvrd, buffermgrd, natmgr, natsync – waiting for PortInitDone
+
+## Init types:
+The Dynamic port add/remove configuration will be supported for both types of init types:<br />
+•   Start the system with full ports on config db <br />
+•   Start the system without some of the ports on config db<br />
+•   Start the system with zero ports on config db<br />
+
+**Note:** This is a new type of init that was never tested and will be supported.<br />
+The zero-port system is a special case of this feature. <br />
+Few PRs were already added in order to support zero ports init.<br />
+
+after init stage we can add/remove ports dynamically through redis call to add/remove entry to/from port table on config db ("PORT")
+
+## Init with zero ports:
+Starting with zero ports requires new SKU for zero ports with these changes:<br />
+**Port_confg.ini** – without entries<br />
+**Hwsku.json** – without interfaces<br />
+**Platform.json** – without interfaces<br />
+**Sai xml** file – needs to be without port entries. <br />
```

Comment:
I will change it to sai profile
on our sai profile we have all the ports configured, but probably in the future, I will consult with the team and it will be changed.